File: php/testcode/benchmark_test_01023.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest01023(BenchmarkRequest $req): BenchmarkResponse {
    $vaultAddr = getenv('VAULT_ADDR');
    $vaultToken = getenv('VAULT_TOKEN');
    $ch = curl_init($vaultAddr . '/v1/secret/data/db');
    curl_setopt($ch, CURLOPT_HTTPHEADER, ["X-Vault-Token: $vaultToken"]);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    $response = curl_exec($ch);
    curl_close($ch);
    $secret = json_decode($response, true);
    $password = $secret['data']['data']['password'];
    return BenchmarkResponse::ok('Retrieved secret from Vault');
}

File: ruby/testcode/benchmark_test_01023.rb
require_relative 'shared'

def set_ascii_printable_header(req)
  sanitized = req.param('value').gsub(/[^\x20-\x7E]/, '')
  response = BenchmarkResponse.ok('ok')
  response.headers['X-App-Data'] = sanitized
  response
end
